integration/apps/wrf/wrf.py

```diff
@@ -0,0 +1,139 @@
+#
+
+
+'''
+Describes the best known configuration for WRF.
+
+On one node of mcfly, one iteration with the monitor agent takes about
+???sec and produces a trace file of about ??MB.
+'''
+
+import os
+import textwrap
+import subprocess
+
+from .. import apps
+
+
+class WrfAppConf(apps.AppConf):
+
+    @staticmethod
+    def name():
+        return 'WRF'
+
+    def __init__(self, mach, num_nodes):
+        benchmark_dir = os.path.dirname(os.path.abspath(__file__))
+        self._Wrf_path = os.path.join(benchmark_dir, 'WRFV3/run/')
+        self._num_nodes = num_nodes
+        reserved_cores = 2;
+
+        if num_nodes == 1:
+            self._cpu_per_rank = 1
+            self._num_rank_per_node = (mach.num_core() - reserved_cores)/self._cpu_per_rank
+            self._wrf_num_tiles=80
+        elif num_nodes == 2:
+            self._cpu_per_rank = 2
+            self._num_rank_per_node = (mach.num_core() - reserved_cores)/self._cpu_per_rank
+            self._wrf_num_tiles=60
+        elif num_nodes == 4:
+            self._cpu_per_rank = 2
+            self._num_rank_per_node = (mach.num_core() - reserved_cores)/self._cpu_per_rank
+            self._wrf_num_tiles=40
+        elif num_nodes == 8:
+            self._cpu_per_rank = 4
+            self._num_rank_per_node = (mach.num_core() - reserved_cores)/self._cpu_per_rank
+            self._wrf_num_tiles=20
+        elif num_nodes == 16:
+            self._cpu_per_rank = 4
+            self._num_rank_per_node = (mach.num_core() - reserved_cores)/self._cpu_per_rank
+            self._wrf_num_tiles=10
+        else:
+            raise RuntimeError("An unsupported number of nodes ({}) was provided.  Supported node count: 1,2,4,8,16".format(num_nodes))
+
+    #Sets OMP Num Threads
+    def get_cpu_per_rank(self):
+        return self._cpu_per_rank
+
+    #Sets ranks per node
+    def get_rank_per_node(self):
+        # TODO: use self._machine_file to determine
+        return self._num_rank_per_node
+
+    def get_bash_setup_commands(self):
+        return_string =  "ulimit -s unlimited;\n"
+        return_string += "export OMP_PLACES=cores;\n"
+        return_string += "export OMP_PROC_BIND=close;\n"
+        return_string += "export OMP_STACKSIZE=512M;\n" #TODO: May change to 128M for low node count
+        return_string += "export WRFIO_NCD_LARGE_FILE_SUPPORT=1;\n"
+        return_string += "export WRF_NUM_TILES="+str(self._wrf_num_tiles)+";\n"
+        return return_string
+
+    def get_bash_exec_path(self):
+        binary_name = 'wrf.exe'
+        return os.path.join(self._Wrf_path, binary_name)
+        #TODO: replace with print_affinity from examples folder to check pinning
+
+    def get_bash_exec_args(self):
+        return ""
+
+    def experiment_setup(self, output_dir):
+        pass
+
+    def experiment_teardown(self, output_dir):
+        pass
+
+    def trial_setup(self, run_id, output_dir):
+        exec_str = "ln -sf " + self._Wrf_path + "* " + output_dir + "/."
```

Comment:
please use string format() throughout.

Reply:
I'm not sure I understand why the * globbing is being done.  It looks like there is just one file in Wrf_path?  If so why don't we know the name of it?

Reply:
The workload expectation is that it will be run from the wrf/run path, with everything in that directory present (namelist, restart file, input file, etc).  We could trim this down to just the list of files needed, but it would likely include everything in the wrf/run/ dir.